File: 033. First Duplicate Value/FirstDuplicateValue.java
import java.util.*;

class FirstDuplicateValue {

    public int firstDuplicateValue(int[] array) {
        // Write your code here.
        HashSet<Integer> seen = new HashSet<Integer>();
        for (int num : array) {
            if (seen.contains(num))
                return num;
            seen.add(num);
        }
        return -1;
    }
}

File: 033. First Duplicate Value/FirstDuplicateValue.kt
package com.algoexpert.program

fun firstDuplicateValue(array: MutableList<Int>): Int {
    val seen = mutableSetOf<Int>()
    for (num in array) {
        if (num in seen) return num
        seen.add(num)
    }
    return -1
}
